tests/cnf/core/network/internal/netenv/netenv.go

```
@@ -234,3 +238,103 @@ func RemoveAllPoliciesAndWaitForSriovAndMCPStable() error {
+
+		return nil, fmt.Errorf("insufficient speaker IP addresses: got %d, need at least %d",
+			len(nextHopList), len(podList))
+	}
+
+	routesMap := make(map[string]string)
+
+	for _, frrPod := range podList {
+		if frrPod.Definition.Spec.NodeName == workerNodeList[0].Definition.Name {
+			routesMap[frrPod.Definition.Spec.NodeName] = nextHopList[1]
+		} else {
+			routesMap[frrPod.Definition.Spec.NodeName] = nextHopList[0]
+		}
+	}
+
+	return routesMap, nil
+}
+
+// SetStaticRoute could set or delete static route on all Speaker pods.
+func SetStaticRoute(frrPod *pod.Builder, action, destIP, containerName string,
+	nextHopMap map[string]string) (string, error) {
+	buffer, err := frrPod.ExecCommand(
+		[]string{"ip", "route", action, destIP, "via", nextHopMap[frrPod.Definition.Spec.NodeName]}, containerName)
+	if err != nil {
+		if strings.Contains(buffer.String(), "File exists") {
+			glog.V(90).Infof("Warning: Route to %s already exist", destIP)
+
+			return buffer.String(), nil
+		}
+
+		if strings.Contains(buffer.String(), "No such process") {
+			glog.V(90).Infof("Warning: Route to %s already absent", destIP)
+
+			return buffer.String(), nil
+		}
+
+		return buffer.String(), err
+	}
+
+	return buffer.String(), nil
+}
+
+// CreateExternalNad creats an external network-attchment-definition using the br-ex interface.
+func CreateExternalNad(apiClient *clients.Settings, name, testNameSpace string) {
+	glog.V(2).Info("Creating external BR-EX NetworkAttachmentDefinition")
+
+	// Define the master NAD plugin
+	macVlanPlugin, err := define.MasterNadPlugin(coreparams.OvnExternalBridge, "bridge", nad.IPAMStatic())
+	if err != nil {
+		glog.V(90).Infof("Failed to define master NAD plugin: %v", err)
+
+		return
+	}
+
+	// Create the NetworkAttachmentDefinition
+	_, err = define.CreateNadWithMasterPlugin(apiClient, name, testNameSpace, macVlanPlugin)
+	if err != nil {
+		glog.V(90).Infof("Failed to create external NetworkAttachmentDefinition: %v", err)
+
+		return
```

Please return the error

---

?